tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/impl/PostServiceImpl.java

```diff
@@ -137,6 +141,10 @@ public PostDTO save(PostDTO postDTO) {
     Post post = postMapper.postDTOToPost(postDTO);
     post = postRepository.save(post);
     handleNewPostEsrNotification(postDTO);
+    if (postDTO.getId() != null) {
```

Comment:
How do I determine the difference between a new DTO and a saved DTO? (I don't think it matters in the end because it's the same result, but they're technically different events which might matter in future)

Reply:
Never mind, stupid question, there's two different methods 🙈

Reply:
https://github.com/Health-Education-England/TIS-TCS/blob/03cde00531cf3f9f3a4df018cb54d846ddd50bf5/tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/PostResource.java#L118
The postDTO we receive from the frontend are not allowed to have an ID, so I think here, you will never publish this PostSavedEvent.